### PR TITLE
DESIGN - authn/cli ux - require explicit config of CA roots

### LINKED ISSUE #5: rework CLI UX to force explicit configuration of CA roots used for client cert validation
### Issue:

> how users [of the load balancer] will configure which CAs to trust vs not trust for client authentication?
> If trusted CAs are loaded from "the usual operating-system defined location" and if only a subset of those certificates are trusted to authenticate clients, how will the server know which CAs to trust and which not to?

raised by timothyb89 during review https://github.com/fcostin/tcplb/pull/2/files#r871890760

### Decision:

rework CLI UX to force the user of the server to specify a custom location of the CA roots used to validate the client's certificate. that adds a little friction but makes the decision explicit and is safer.

### To Do:

- [ ] update CLI UX section of design doc
- [ ] update authentication section of design doc

### PR DESCRIPTION
**Pull request by fcostin**
Rework CLI UX to force the user of the server to specify a
custom location of the CA roots used to validate the
client's certificate. This adds a little friction to UX
but makes the decision explicit but is safer, as it has
significant security consequences.

resolves https://github.com/fcostin/tcplb/issues/5

### CODE REVIEW

File: docs/DESIGN.md [hunk 1]
```diff
@@ -222,24 +229,24 @@ The server must be configured with a certificate and corresponding private key.
...
 connections.
 
-The server will load trusted CA certs from the environment in the usual 
-operating-system defined location. These CA certs will be used to validate 
-the trust chain of the certificate presented by the client. Since the client 
-certificate will be used as the basis of authentication, the server MUST NOT 
-be configured to trust CA roots that are not trusted to verify client 
-identities for purposes of authentication.
+The server will load the CA roots to use for client certificate
+validation from an explicit location specified by the user (see CLI UX).
```
Comment (codingllama) [inline docs/DESIGN.md:233]: For the record I think this is better too. It's easier to tighten/reason about the system this way.

File: docs/DESIGN.md [hunk 2]
```diff
@@ -60,11 +60,18 @@ minimal set of options using flags:
...
+  authentication section)
 - list of one or more upstream addresses
 
 Load balancer will display help when invoked with no arguments and exit with 
 nonzero status on fatal errors.
 
+Load balancer will regard the following as fatal errors that prevent startup:
+- no bind address specified
+- certificate & private key not specified or failed to load
+- trusted CA roots not specified or failed to load
```
Comment (codingllama) [inline docs/DESIGN.md:73]: I expect clients will want a similar setting?

Edit: no need to change the spec, just looking for confirmation (or not).

**Review by codingllama — APPROVED** [review]

**Review by timothyb89 — APPROVED** [review]
That clears things up, thanks!

File: docs/DESIGN.md (hunk 2 above)
Comment (fcostin) [reply docs/DESIGN.md:73]: yes. i guess it depends what the intent of the client is. assuming the client wants to connect to our specific load balancer server and are not willing to connect to arbitrary servers, the client could be built in a similar way.

**Pull request merged**